### PR TITLE
Added ssept model path & citation

### PR DESCRIPTION
**Pull request by aeroabir**
### Description
In the examples\sasrec_amazon.ipynb notebook added the path for SSEPT (Personalized Transformer) model. Also added the citation of the same paper. 

### Related Issues
NA.

### Checklist:

- [ x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ x] I have added tests covering my contributions.
- [ x] I have updated the documentation accordingly.
- [ x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: examples/00_quick_start/sasrec_amazon.ipynb [hunk 1]
```diff
@@ -226,16 +217,16 @@
...
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "../../tests/resources/deeprec/sasrec/reviews_Electronics_5.txt\n",
-      "36262 Users and 35074 items\n",
-      "average sequence length: 15.01\n"
+      "20247 Users and 11589 items\n",
```
Comment (anargyri) [inline examples/00_quick_start/sasrec_amazon.ipynb:228]: Why do we have now fewer users and items? Did the data change?

**Review by anargyri — APPROVED** [review]

File: examples/00_quick_start/sasrec_amazon.ipynb (hunk 1 above)
Comment (aeroabir) [reply examples/00_quick_start/sasrec_amazon.ipynb:228]: The source data did not change. I think it was some intermediate file that was loaded earlier while I was testing with the <filter_K_core> function.

Comment (anargyri) [reply examples/00_quick_start/sasrec_amazon.ipynb:228]: So are the current numbers correct?

Comment (miguelgfierro) [reply examples/00_quick_start/sasrec_amazon.ipynb:228]: do you know what is happening with the intermediate file?

File: examples/00_quick_start/sasrec_amazon.ipynb [hunk 2]
```diff
@@ -9,13 +9,13 @@
    "cell_type": "markdown",
```
Comment (miguelgfierro) [inline examples/00_quick_start/sasrec_amazon.ipynb:9]: <pre class="ql-syntax" spellcheck="false">Line #1.    model_name = 'sasrec'  # 'sasrec' or 'ssept'
</pre><p>Minor detail, if we add <code>model_name</code> to the previous cell, which is the one that is configured with papermill,  we could add an integration test for SSEPT, just copying this one <a href="https://github.com/microsoft/recommenders/blob/main/tests/integration/examples/test_notebooks_gpu.py#L622" target="_blank">recommenders/test_notebooks_gpu.py at main · microsoft/recommenders (github.com)</a> and updating the outputs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1678/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (miguelgfierro) [inline examples/00_quick_start/sasrec_amazon.ipynb:9]: <p>maybe this function could go in the libraries, and we can add a small comment here mentioning that we are filtering users and items</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1678/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: examples/00_quick_start/sasrec_amazon.ipynb [hunk 3]
```diff
@@ -9,13 +9,13 @@
...
     "\n",
-    "This is a class of sequential recommendation that uses Transformer \\[2\\] for encoding the users preference represented in terms of a sequence of items purchased/viewed before. Instead of using CNN (Caser \\[3\\]) or RNN (GRU4Rec \\[4\\], SLI-Rec \\[5\\] etc.) the approach relies on Transformer based encoder that generates a new representation of the item sequence. This notebook provides an example of necessary steps to train and test a SASRec model. "
+    "This is a class of sequential recommendation that uses Transformer \\[2\\] for encoding the users preference represented in terms of a sequence of items purchased/viewed before. Instead of using CNN (Caser \\[3\\]) or RNN (GRU4Rec \\[4\\], SLI-Rec \\[5\\] etc.) the approach relies on Transformer based encoder that generates a new representation of the item sequence. There are two variants of this Transformer based approach included here, (a) based on vanilla Transformer that models only the item sequence, called SASRec \\[1\\] and (b) personalized Transformer, called SSE-PT, \\[6\\]) that also models the users. This notebook provides an example of necessary steps to train and test a SASRec model. "
```
Comment (miguelgfierro) [inline examples/00_quick_start/sasrec_amazon.ipynb:18]: `This notebook provides an example of necessary steps to train and test a SASRec model.`
here we are also using SSE-PT, we could say:
`This notebook provides an example of necessary steps to train and test a SASRec model or a SSE-PT model.`

**Review by miguelgfierro — COMMENTED** [review]
I added some small comments, but overall is great!

File: examples/00_quick_start/sasrec_amazon.ipynb (hunk 1 above)
Comment (aeroabir) [reply examples/00_quick_start/sasrec_amazon.ipynb:228]: The intermediate files are stored in the data directory. 

I think this is correct and same as what we get from daisyRec package ([link](https://github.com/AmazingDD/daisyRec)). 

image

File: examples/00_quick_start/sasrec_amazon.ipynb (hunk 2 above)
Comment (aeroabir) [reply examples/00_quick_start/sasrec_amazon.ipynb:9]: Modified the test to have two parameters, one for SASRec and one for SSEPT.

Comment (aeroabir) [reply examples/00_quick_start/sasrec_amazon.ipynb:9]: Moved to datasets/split_utils.py

File: examples/00_quick_start/sasrec_amazon.ipynb (hunk 3 above)
Comment (aeroabir) [reply examples/00_quick_start/sasrec_amazon.ipynb]: Modified.

File: tests/integration/examples/test_notebooks_gpu.py [hunk 4]
```diff
@@ -608,15 +656,24 @@ def test_cornac_bivae_integration(
...
             {"ndcg@10": 0.2626, "Hit@10": 0.4244},
             42,
-        )
+        ),
+        (
+            os.path.join("tests", "recsys_data", "RecSys", "SASRec-tf2", "data"),
+            1,
+            128,
+            "ssept",
+            {"ndcg@10": 0.2626, "Hit@10": 0.4244},
```
Comment (miguelgfierro) [inline tests/integration/examples/test_notebooks_gpu.py:674]: @aeroabir this is great, the only detail to check is, are you sure that both algos get the same NDCG and HIT?

File: recommenders/datasets/split_utils.py [hunk 5]
```diff
@@ -163,3 +163,35 @@ def split_pandas_data_with_ratios(data, ratios, seed=42, shuffle=False):
...
     return splits
+
+
+def filter_k_core(data, core_num=0, col_user="userID", col_item="itemID"):
+    """Filter rating dataframe for minimum number of users and items by
+    repeatedly applying min_rating_filter until the condition is satisfied.
+
+    """
+    num_users, num_items = len(data[col_user].unique()), len(data[col_item].unique())
+    print(f"Original: {num_users} users and {num_items} items")
```
Comment (miguelgfierro) [inline recommenders/datasets/split_utils.py:174]: minor point can we add
```
import logging
log = logging.getLogger(__name__)
```
as in https://github.com/microsoft/recommenders/blob/main/recommenders/datasets/download_utils.py#L15
and then change the prints with log.info?

**Review by miguelgfierro — COMMENTED** [review]
minor comment

File: tests/integration/examples/test_notebooks_gpu.py (hunk 4 above)
Comment (aeroabir) [reply tests/integration/examples/test_notebooks_gpu.py:674]: @miguelgfierro the algorithms generate different results but the current TOL and ABS_TOL take care of the differences.

File: recommenders/datasets/split_utils.py (hunk 5 above)
Comment (aeroabir) [reply recommenders/datasets/split_utils.py]: Done.

**Pull request merged**